File: CS-Examples/06_Comments/RemoveComment.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Collections;

namespace RemoveComment
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CommentSample.xlsx");

            //Get all comments of the first sheet
            CommentsCollection comments = workbook.Worksheets[0].Comments;

            //Change the content of the first comment
            comments[0].Text = "This comment has been changed.";

            //Remove the second comment
            comments[1].Remove();

            // Specify the resulting file name
            string output = "RemoveAndChangeComment.xlsx";

            // Save the modified workbook to a file using Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/06_Comments/RemoveComment.vb
Imports Spire.Xls
Imports Spire.Xls.Collections

Namespace RemoveComment

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CommentSample.xlsx")

            ' Retrieve the CommentsCollection from the first worksheet of the workbook
            Dim comments As CommentsCollection = workbook.Worksheets(0).Comments

            ' Modify the text of the first comment
            comments(0).Text = "This comment has been changed."

            ' Remove the second comment from the collection
            comments(1).Remove()

            ' Specify the output file name for saving the modified workbook
            Dim output As String = "RemoveAndChangeComment.xlsx"

            ' Save the workbook to the specified file path using Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
